src/main/kotlin/anttracker/PageOf.kt

```diff
@@ -0,0 +1,119 @@
+package anttracker
+import org.jetbrains.exposed.sql.SizedIterable
+import org.jetbrains.exposed.sql.transactions.transaction
+import kotlin.math.ceil
+
+// A PageOf objects contains a max-20 element list of T.
+// Provides top-level functionality for:
+//  - loading DB records into memory (one page at a time)
+//  - accessing records
+//  - displaying records as a page
+//  - going to the next page and loading its records
+abstract class PageOf<IntEntity> {
```

Comment:
I think it would be better to pass the total number of records to the base class `PageOf` and have it calculate the total number of pages.

It could look like this.

```kotlin
abstract class PageOf<IntEntity>(
    private val totalRecords: Long,
) {
    private val lastPageNum: Long = calculateTotalPages(totalRecords)
```

Reply:
yup, agreed. done

Reply:
I agree it'd be nice for the base class to do it, but this calc (at least for Releases) is dependent on props of the subclass, which inits AFTER base class initialization; this leads to props being null during calculation.